test/test_storage.py

```diff
@@ -312,17 +318,26 @@ def upload_progress(progress):
         st.delete_key(key)
 
 
+# sftp test support is available in vagrant, so can test just like for local if running in vagrant
 def _test_storage_init(storage_type, with_prefix, tmpdir, config_overrides=None):
     if storage_type == "local":
         storage_config = {"directory": str(tmpdir.join("rohmu"))}
+    elif storage_type == 'sftp' and os.path.isfile('/home/vagrant/pghoard-test-sftp-user'):
+        with open('/home/vagrant/pghoard-test-sftp-user', 'r') as sftpuser:
+            username, password = sftpuser.read().strip().split(':')
+
+        storage_config = {'server': 'localhost', 'port': 22, 'username': username, 'password': password}
+
+        # for no prefix testing, we need to cleanup existing files
+        os.system('sudo rm -rf /home/' + username + '/*')
```

Comment:
Couple of nitpicks: 1) `"` is used all around this new block so that should be used instead of `'` and `.format()` is nicer than `+`.

And more importantly: I would add something that bails out in case `username` is empty or else really bad things may happen here...

Reply:
ok, I changed the whole file to use single quotes, even though most of it I did not touch, and I won't rm if username is not defined.

Reply:
Sorry you misunderstood my comment. I meant this block should be using double quotes as well because that's what's predominantly used in pghoard codebase otherwise.

Reply:
ok will change it back and get rid of the additional commit